girder/api/v1/item.py

```diff
@@ -172,7 +175,32 @@ def setMetadata(self, item, metadata, params):
                 raise RestException(
                     'Invalid key %s: keys must not start with the "$" character.' % k)
 
-        return self.model('item').setMetadata(item, metadata)
+        return self.model('item').setMetadata(item, metadata, params['allowNull'])
+
+    @autoDescribeRoute(
+        Description('Delete metadata fields on an item.')
+        .responseClass('Item')
+        .modelParam('id', model='item', level=AccessType.WRITE)
+        .jsonParam('fields', 'A JSON list containing the metadata fields to delete',
+                   paramType='body')
+        .errorResponse(('ID was invalid.',
+                        'Invalid JSON passed in request body.',
+                        'Metadata key name was invalid.'))
+        .errorResponse('Write acess was denied for the item.', 403)
+    )
+    def deleteMetadata(self, item, fields, params):
+        if not fields:
+            raise RestException('Key names must not be empty.')
+
+        for k in fields:
+            if '.' in k:
+                raise RestException(
+                    'Invalid key %s: keys must not contain the "." character.' % k)
+            if k[0] == '$':
+                raise RestException(
+                    'Invalid key %s: keys must not start with the "$" character.' % k)
```

Comment:
Since this logic exists twice in this file, we should move it to a helper function.